### PR TITLE
Test cleanup

### PR DESCRIPTION
**Pull request by ursjoss**
* Reducing package access modifiers (junit5)
* convert some anonymous classes to lambdas
* use diamond operators where possible

### CODE REVIEW

File: bootstrap-core/src/test/java/de/agilecoders/wicket/core/WicketApplicationTest.java [hunk 1]
```diff
@@ -83,11 +83,11 @@ protected void init(WebApplication app) {}
     protected void onBefore() {
     }
 
-    protected Class<? extends Page> getHomePage() {
+    private Class<? extends Page> getHomePage() {
         return Page.class;
     }
 
-    protected IBootstrapSettings createBootstrapSettings() {
+    private IBootstrapSettings createBootstrapSettings() {
```
Comment (martin-g) [inline bootstrap-core/src/test/java/de/agilecoders/wicket/core/WicketApplicationTest.java]: those are `protected` intentionally. This way it is easier for a test to override whatever is needed.

File: bootstrap-extensions/src/test/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/WicketApplicationTest.java [hunk 2]
```diff
@@ -58,29 +48,16 @@ protected void init() {
     protected void onBefore() {
     }
 
-    protected Class<? extends Page> getHomePage() {
+    private Class<? extends Page> getHomePage() {
         return Page.class;
     }
 
-    protected IBootstrapSettings createBootstrapSettings() {
+    private IBootstrapSettings createBootstrapSettings() {
```
Comment (martin-g) [inline bootstrap-extensions/src/test/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/WicketApplicationTest.java]: Same as earlier

**Review by martin-g — CHANGES_REQUESTED** [review]
Please revert the changes related to `protected` visibility in **ApplicationTest classes.

What bothers me in this PR is that it will be hard to merge to the newer branches (Bootstrap 4.x and Wicket 9.x ones). Usually such kind of improvements I make in the Wicket.NEXT branch and I do not bother to downport them to older ones.

**Comment by ursjoss** [comment]
@martin-g Yeah, you're right. I am currently only using the `wicket-8.x` branch. However, I don't mind applying the equivalent changes to `wicket-9.x` and `wicket-8.x-bootstrap-4.x` if that make sense. Just let me know.

**Comment by ursjoss** [comment]
I applied the requested changes to this PR.

**Comment by martin-g** [comment]
@ursjoss Please merge these changes to `wicket-9.x` and `wicket-8.x-bootstrap-4.x`! Thanks!

**Comment by ursjoss** [comment]
@martin-g Ok, I'll follow up with two separate PRs. Thanks for merging.

**Comment by martin-g** [comment]
I know that most of the users of Wicket-Bootstrap library use Wicket 8.x + Bootstrap 3.x. And most of the PRs here are for this branch. But I port the fixes to the newer branches by running `git checkout wicket-9.x; git merge wicket-8.x` from time to time. This leads to merge conflicts in the pom.xml files but once resolved the Git history is in tact and next time it is easier to merge again.

**Pull request merged**